### ISSUE TITLE
Tag (tag) component has insufficient contrast on Gray 100 theme

### ISSUE THREAD

**Issue by jsrilapan** [carbon-design-system/carbon#4022]
## Environment

Windows 10
Chrome Version 76.0.3809.132 (Official Build) (64-bit)

## Detailed description

Carbon v10 - React

Component: Tag (filter), http://themes.carbondesignsystem.com/?nav=tag

The component is using,
`background-color: interactive-02, color: inverse-01`

When switching to the Gray 100 theme, the contrast ratio has dropped to 3.57 (WCAG AA: Fail).
I used WebAIM (https://webaim.org/resources/contrastchecker/) to check the colour contrast.

## Additional information

![tag](https://user-images.githubusercontent.com/51158937/65159619-4c001200-da2c-11e9-87e8-0f764dacedb0.png)

**Comment by asudoh** [comment]
@carbon-design-system/design Any thoughts...? Thanks!

**Comment by jsrilapan** [comment]
According to the High Contrast section in the color guideline (https://www.carbondesignsystem.com/guidelines/color/overview), it suggests that _$inverse-02_ is the correct token for background colour.

Example,
component: Tag (filter)
theme: Gray 100
style: `background-color: inverse-02, color: inverse-01`
contrast: 16.15

![Annotation 2019-09-19 094529](https://user-images.githubusercontent.com/51158937/65228463-6e446f00-dac2-11e9-8bb2-94f5809c0c15.png)

**Comment by laurenmrice** [comment]
Yes, this is a bug in code for the dark themes. It should be:

Tag color: `inverse-02`
Close icon color:  `inverse-01`
Close icon hover color: `inverse-hover-ui`
Text color: `inverse-01`
